extensions/ratelimiting.md

```diff
@@ -0,0 +1,117 @@
+| --- | --- | --- | --- |
+| extensionid | Unique extension Id (default is 'workflow-kpi-extension') | string | yes |
+| workflowid | Workflow definition unique identifier (workflow id property) | string | yes |
+| workflowVersions | Workflow versions. If not defined, applies to all workflow instances (regardless of their associated workflow version) | array | no |
+| [singleInstance](#Single-Instance-Definition) | Rate limits per single workflow instance | object | no |
+| [allInstances](#All-Instances-Definition) | Rate limits per all workflow instances | object | no |
+
+### Single Instance Definition
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| maxActionsPerSecond | Sets the rate limiting on number of actions that can be executed per second. Notice that the number is represented as number type, so that you can set it to less than 1 if needed. For example, set the number to 0.1 means you want workflow actions should be executed once every 10 seconds. Default zero value means 'unlimited'| number | no |
+| maxConcurrentActions | Maximum number of actions that can be executed in parallel | string | no |
+| maxProducedEventsPerSecond |Sets the rate limiting on number of events that can be produced per second. Notice that the number is represented as number type, so that you can set it to less than 1 if needed. For example, set the number to 0.1 means workflow can produce events once every 10 seconds. Default zero value means 'unlimited' | string | no |
+| maxStates | Maximum number of workflow states that should be executed. Default is zero, meaning unlimited. | string | no |
+| maxTransitions | Maximum number of workflow transitions that should be executed. Default is zero, meaning unlimited. | string | no |
+
+### All Instances Definition
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| maxActionsPerSecond | Sets the rate limiting on number of actions that can be executed per second. Notice that the number is represented as number type, so that you can set it to less than 1 if needed. For example, set the number to 0.1 means you want workflow actions should be executed once every 10 seconds. Default zero value means 'unlimited'| number | no |
+| maxConcurrentActions | Maximum number of actions that can be executed in parallel | string | no |
+| maxProducedEventsPerSecond |Sets the rate limiting on number of events that can be produced per second. Notice that the number is represented as number type, so that you can set it to less than 1 if needed. For example, set the number to 0.1 means workflow can produce events once every 10 seconds. Default zero value means 'unlimited' | string | no |
+| maxStates | Maximum number of workflow states that should be executed. Default is zero, meaning unlimited. | string | no |
+| maxTransitions | Maximum number of workflow transitions that should be executed. Default is zero, meaning unlimited. | string | no |
```

Comment:
```suggestion
| maxActionsPerSecond | Sets the rate-limiting on a number of actions that can be executed per second. Notice that the number is represented as number type so that you can set it to less than 1 if needed. For example, set the number to 0.1 means you want workflow actions should be executed once every 10 seconds. Default zero value means 'unlimited'| number | no |
| maxConcurrentActions | Maximum number of actions that can be executed in parallel | string | no |
| maxProducedEventsPerSecond | Sets the rate-limiting on a number of events that can be produced per second. Notice that the number is represented as number type so that you can set it to less than 1 if needed. For example, set the number to 0.1 means workflow can produce events once every 10 seconds. Default zero value means 'unlimited' | string | no |
| maxStates | Maximum number of workflow states that should be executed. Default is zero, meaning unlimited. | string | no |
| maxTransitions | The maximum number of workflow transitions that should be executed. Default is zero, meaning unlimited. | string | no |
```